src/docs/development/add-to-app/ios/project-setup.md

```diff
@@ -175,45 +175,59 @@ some/path/MyApp/
     │   ├── App.framework
     │   ├── FlutterPluginRegistrant.framework
     │   └── example_plugin.framework (each plugin with iOS platform code is a separate framework)
-      ├── Profile/
-      │   ├── Flutter.framework
-      │   ├── App.framework
-      │   ├── FlutterPluginRegistrant.framework
-      │   └── example_plugin.framework
-      └── Release/
-          ├── Flutter.framework
-          ├── App.framework
-          ├── FlutterPluginRegistrant.framework
-          └── example_plugin.framework
+    ├── Profile/
+    │   ├── Flutter.framework
+    │   ├── App.framework
+    │   ├── FlutterPluginRegistrant.framework
+    │   └── example_plugin.framework
+    └── Release/
+        ├── Flutter.framework
+        ├── App.framework
+        ├── FlutterPluginRegistrant.framework
+        └── example_plugin.framework
 ```
 
 {{site.alert.tip}}
-With Xcode 11 installed, you can generate [XCFrameworks][] instead of universal frameworks by adding
-the flags `--xcframework --no-universal`.
+  With Xcode 11 installed, you can generate [XCFrameworks][] instead of universal frameworks by adding
+  the flags `--xcframework --no-universal`.
 {{site.alert.end}}
 
-Embed the generated frameworks into your existing application in Xcode. For example, you can
-drag the frameworks from `some/path/MyApp/Flutter/Release/` in Finder
-into your targets's build settings > General > Frameworks, Libraries, and Embedded Content. Then, select
-"Embed & Sign" from the drop-down list.
+Embed and link the generated frameworks into your existing application in Xcode.
+There are multiple ways to do this—use the method that is best for your project.
 
-{% include app-figure.md image="development/add-to-app/ios/project-setup/embed-xcode.png" alt="Embed frameworks in Xcode" %}
+### Link on the frameworks
+For example, you can drag the frameworks from `some/path/MyApp/Flutter/Release/` in Finder
+into your targets' build settings > Build Phases > Link Binary With Libraries.
 
-In the target's build settings, add `$(PROJECT_DIR)/Flutter/Release/` to your Framework Search Paths (`FRAMEWORK_SEARCH_PATHS`).
+In the target's build settings, add `$(PROJECT_DIR)/Flutter/Release/` to the **Framework Search Paths** (`FRAMEWORK_SEARCH_PATHS`).
 
 {% include app-figure.md image="development/add-to-app/ios/project-setup/framework-search-paths.png" alt="Update Framework Search Paths in Xcode" %}
 
-There are multiple ways to embed frameworks into an Xcode project—use the method that is best for your project.
+### Embed the frameworks
+The generated dynamic frameworks must be embedded into your app to be loaded at runtime.
+
+{{site.alert.important}}
+  Plugins may define themselves as static or dynamic. Static frameworks should be linked on, but
+  never embedded. If you embed a static framework into your application, your application is not
+  publishable to the App Store and fails with a **Found an unexpected Mach-O header code** archive error.
```

Comment:
What tells me if the library is static or dynamic?

Reply:
Good question, I was trying to think of the most succinct way to describe this.  All of these options seem pretty advanced.
Method 1:
Grep podspecs for plugins opting into static.
```
$  grep -rl "static_framework" .ios/Pods | xargs -0 basename
FlutterPluginRegistrant.podspec.json
```
Method 2:
otool shows filetype OBJECT instead of DYLIB.
```
$ otool -h -v <output>/Release/FlutterPluginRegistrant.framework/FlutterPluginRegistrant
...
Mach header
      magic cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
   MH_MAGIC     ARM         V7  0x00      OBJECT    20       2032 SUBSECTIONS_VIA_SYMBOLS
...
$ otool -h -v <output>/Flutter/Release/device_info.framework/device_info 
Mach header
      magic cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
   MH_MAGIC     ARM         V7  0x00       DYLIB    22       2608   NOUNDEFS DYLDLINK TWOLEVEL NO_REEXPORTED_DYLIBS
...
```
Method 3:
Check Pods Xcode project build settings for MACH_O_TYPE.
```
$ cd .ios/Pods
$ $ /usr/bin/xcodebuild -alltargets -showBuildSettings | egrep "(MACH_O_TYPE|FULL_PRODUCT_NAME)"
    FULL_PRODUCT_NAME = device_info.framework
    MACH_O_TYPE = mh_dylib
    FULL_PRODUCT_NAME = FlutterPluginRegistrant.framework
    MACH_O_TYPE = staticlib
    FULL_PRODUCT_NAME = path_provider.framework
    MACH_O_TYPE = mh_dylib
    FULL_PRODUCT_NAME = Pods_Runner.framework
    MACH_O_TYPE = staticlib
    FULL_PRODUCT_NAME = device_info.framework
    MACH_O_TYPE = mh_dylib
```

These are all pretty bad for new developers...  grepping for `static_framework` (Method 1) seems like it would be the easiest to understand
Ideally, when you build the ios-frameworks the tool will warn which generated frameworks NOT to embed.

Reply:
Sounds good.  Maybe even just link to this SO post? https://stackoverflow.com/questions/32591878/ios-is-it-a-static-or-a-dynamic-framework/32591983

That just uses the `file` utility.